python/paddle/distribution/exponential.py

```diff
@@ -0,0 +1,164 @@
+        self.rate = rate
+        super().__init__(self.rate.shape)
+
+    @property
+    def mean(self):
+        """Mean of exponential distribuion.
+
+        Returns:
+            Tensor: mean value.
+        """
+        return self.rate.reciprocal()
+
+    @property
+    def variance(self):
+        """Variance of exponential distribution.
+
+        Returns:
+            Tensor: variance value.
+        """
+        return self.rate.pow(-2)
+
+    def sample(self, shape=()):
+        """Generate samples of the specified shape.
+
+        Args:
+            shape (Sequence[int], optional): Shape of the generated samples.
+
+        Returns:
+            Tensor, A tensor with prepended dimensions shape.The data type is float32.
+        """
+        with paddle.no_grad():
+            return self.rsample(shape)
+
+    def rsample(self, shape=()):
+        """Generate reparameterized samples of the specified shape.
+
+        Args:
+            shape (Sequence[int], optional): Shape of the generated samples.
+
+        Returns:
+            Tensor: A tensor with prepended dimensions shape.The data type is float32.
+        """
+        shape = distribution.Distribution._extend_shape(
+            self, sample_shape=shape
+        )
+
+        uniform = paddle.uniform(
+            shape=shape,
+            min=float(np.finfo(dtype='float32').tiny),
+            max=1.0,
+            dtype=self.rate.dtype,
+        )
+
+        return -paddle.log(uniform) / self.rate
+
+    def prob(self, value):
+        """Probability density funciotn evaluated at value
+
+        Args:
+            value (Tensor): Value to be evaluated.
+
+        Returns:
+            Tensor: Probability.
+        """
+        return self.rate * paddle.exp(-self.rate * value)
+
+    def log_prob(self, value):
+        """Log probability density function evaluated at value
+
+        Args:
+            value (Tensor): Value to be evaluated
+
+        Returns:
+            Tensor: Log probability.
+        """
+        return paddle.log(self.rate) - self.rate * value
+
+    def entropy(self):
+        """Entropy of exponential distribution
+
+        Returns:
+            Tensor: Entropy.
+        """
+        return 1.0 - paddle.log(self.rate)
+
+    def kl_divergence(self, other):
+        """The KL-divergence between two exponential distributions.
+
+        Args:
+            other (Exponential): instance of Exponential.
+
+        Returns:
+            Tensor: kl-divergence between two exponential distributions.
+        """
+        if not isinstance(other, Exponential):
+            raise TypeError(
+                f"Expected type of other is Exponential, but got {type(other)}"
+            )
+
+        rate_ratio = other.rate / self.rate
+        t1 = -paddle.log(rate_ratio)
+        return t1 + rate_ratio - 1
+
+    @property
+    def _natural_parameters(self):
+        return (-self.rate,)
+
+    def _log_normalizer(self, x):
+        return -paddle.log(-x)
```

Comment:
shall we also add method of `cdf` and `icdf` ?

Reply:
`cdf` and `icdf` have been added in `Exponential`.
In `Gamma`, we need more function APIs to implement them, like `gammainc`.